apps/devhub/tasks.py

```diff
@@ -51,6 +54,20 @@ def validator(upload_id, **kw):
         raise
 
 
+def skip_signing_warning(result_json):
+    """Remove the "Package already signed" warning if we're not signing."""
+    try:
+        result = json.loads(result_json)
+        messages = result['messages']
+    except (KeyError, ValueError):
+        return result_json
+    messages = [m for m in messages if 'signed_xpi' not in m['id']]
+    if len(messages) != len(result['messages']):  # We did remove a warning.
+        result['messages'] = messages
+        result['warnings'] -= 1
```

Comment:
I could be wrong, but I suspect that there may be multiple `signed_xpi` warnings already, for multi-package XPIs.